projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx

```diff
@@ -26,19 +27,21 @@ const {
  * @param {string} props.productSlug  - The WordPress product slug.
  * @param {string} props.redirectUrl  - The URI to redirect to after checkout.
  * @param {string} [props.siteSuffix] - The site suffix.
+ * @param {Function} props.checkSiteHasWpcomProduct - The function used to check whether the site already has the requested product. This will be checked after registration and the checkout page will be skipped if it returns true.
  * @returns {Function}				  - The useEffect hook.
  */
 export default function useProductCheckoutWorkflow( {
 	productSlug,
 	redirectUrl,
 	siteSuffix = defaultSiteSuffix,
+	checkSiteHasWpcomProduct = null,
```

Comment:
have you considered adding this function straight here?
I think the trickier part is the endpoint implementation.

Reply:
minor suggestion. Could we use a nameHandler shape? Something like `siteProductAvailabilityHandler`, or something like that?

Reply:
> have you considered adding this function straight here?
I think the trickier part is the endpoint implementation.

Reply:
> have you considered adding this function straight here?
I think the trickier part is the endpoint implementation.

Good question 👍 . I did consider adding the function here. And you are right, adding the endpoints would be quite tricky for general supporting of all the products, which I believe it's better to leave to you peeps. But anyways, an function to override this would be needed to maximize the flexibility of the component, so I chose only to implement this.

Reply:
> minor suggestion. Could we use a nameHandler shape? Something like siteProductAvailabilityHandler, or something like that?

Addressed in 5328909d19